transports/jms/src/main/java/org/mule/transport/jms/JmsMessageDispatcher.java

```diff
@@ -276,6 +280,35 @@ private MuleMessage dispatchMessage(MuleEvent event, boolean doSend) throws Exce
         }
     }
 
+    /**
+     * Resolve the value of correlationID that should be used for the JMS Message.  This is done here and not as part of
+     * transformation because of the need for visibility of MuleEvent and OutboundEndpoint.
+     *
+     * @param event the current MuleEvent
+     */
+    protected String resolveJmsCorrelationId(MuleEvent event) throws JMSException
+    {
+        return resolveMuleCorrelationId(event);
```

Comment:
This is because other implementations of JMS may use different values for MULE_CORRELATION_ID and JMSMCorrelationID.